### DISCUSSION TITLE
How to get roles with existing users?

### DISCUSSION THREAD

**Discussion by Enhed** [directus/directus#16178] [Archive]
Help pls with filter

**Comment by br41nslug** [comment]
Can you elaborate on what you want to fetch, in what situation?

**Reply by Enhed** [reply]
I wrote about roles for example. I need to get records that have related records o2m type (as users in roles)

**Reply by br41nslug** [reply]
Have you tried filtering against it? when the related count is more than 0 for instance https://docs.directus.io/reference/query.html#array-functions

**Reply by Enhed** [reply]
its work in admin, but in user role not working :( with access fields

**Reply by br41nslug** [reply]
How do you mean in user role? Can you describe in a little more detail what you are trying and how it's not working?

**Reply by Enhed** [reply]
user role without `admin-access`, but have access to all fields in records. User can use `count` in fields:  
<img width="593" alt="image" src="https://user-images.githubusercontent.com/36981278/198001344-cbaf8385-3ad0-4eec-ac49-3ae2719b31dd.png">

but cant use in filter:  
<img width="587" alt="image" src="https://user-images.githubusercontent.com/36981278/198001404-66b6908c-63aa-44bc-9ca2-670907b5d7f2.png">

same request with admin token is working:  
<img width="569" alt="image" src="https://user-images.githubusercontent.com/36981278/198001587-59f42a2e-711e-4d92-85db-0efd8d0fbbd7.png">

**Reply by br41nslug** [reply]
You may be running into this issue https://github.com/directus/directus/issues/16148 for which i've just created a PR https://github.com/directus/directus/pull/16179
